### PR TITLE
EVG-6483 pause task dispatching

### PR DESCRIPTION
**Pull request by ybrill**
When Artifactory went down there was no way to prevent the dispatching of tasks which we knew would fail without turning off task dispatching for all of Evergreen.
This PR is to add an option to projects to turn off task dispatching for just that project's tasks. Since we won't dispatch them, also prevent them from being scheduled by excluding them from the various task finders.

Screen Shot 2020-07-14 at 10 26 12 AM

### CODE REVIEW

**Comment by ablack12** [comment]
Why does disabling the project not work?

**Comment by ybrill** [comment]
Disabling the project would accomplish the same thing, but there are situations where it would have undesired side-effects. For instance, disabled projects won't appear in the project dropdown (which could be confusing for users), users will not be able to submit patches, and new commits won't get picked up by the waterfall.

Adding a granular option to just stop dispatching a project's tasks will let new work enter the system so when the issue has been resolved they will be ready to go.

File: model/task/task.go [hunk 1]
```diff
@@ -1969,7 +1969,8 @@ func FindRunnable(distroID string, removeDeps bool) ([]Task, error) {
 
 	filterDisabledProjects := bson.M{
 		"$match": bson.M{
-			"project_ref.0." + "enabled": true,
+			"project_ref.0." + "enabled":              true,
```
Comment (ablack12) [inline model/task/task.go:1972]: nit: should this use bsonutil.GetDottedKeyName?

File: service/api_task_test.go [hunk 2]
```diff
@@ -204,6 +204,19 @@ func TestAssignNextAvailableTaskWithDispatcherSettingsVersionLegacy(t *testing.T
 			So(h.RunningTask, ShouldEqual, "task1")
 
 		})
+		Convey("tasks with a disabled project should be removed from the queue", func() {
+			pref.Enabled = false
```
Comment (ablack12) [inline service/api_task_test.go:208]: Can this test also test pref.PatchingDisabled = true and pRef.Enabled = true?

File: scheduler/task_finder_test.go [hunk 3]
```diff
@@ -218,6 +218,14 @@ func (s *TaskFinderComparisonSuite) SetupSuite() {
 
 	s.NoError(ref.Insert())
 
+	ref = &model.ProjectRef{
+		Identifier:          "dispatching-disabled",
+		DispatchingDisabled: true,
+		Enabled:             true,
+	}
```
Comment (ablack12) [inline scheduler/task_finder_test.go:225]: I'm confused how the tests in this file work. Does this actually test that tasks aren't able to be dispatched for this project ref, or is it something else?

File: model/task/task.go (hunk 1 above)
Comment (ybrill) [reply model/task/task.go]: done

File: service/api_task_test.go (hunk 2 above)
Comment (ybrill) [reply service/api_task_test.go:208]: done

File: scheduler/task_finder_test.go (hunk 3 above)
Comment (ybrill) [reply scheduler/task_finder_test.go:225]: I think the part of this suite that applies to these changes is the [TestFindRunnableHostsIsIdentical](https://github.com/evergreen-ci/evergreen/blob/84bb055e809b5cb659332c1ecf452e55aa6328ee/scheduler/task_finder_test.go#L269) test. It's making sure each of the 4 methods for finding runnable tasks returns the same list of tasks.

Comment (ybrill) [reply scheduler/task_finder_test.go:247]: I added tests to check tasks aren't returned.

**Pull request merged**